### DISCUSSION TITLE
How to get personalization data for receipt or transaction?

### DISCUSSION THREAD

**Discussion by joey-lott** [etsy/open-api#858] [Q&A] [answered]
When a customer purchases a product from a listing with personalization enabled, how can this data be retrieved via the API? I have an existing app which uses the API to retrieve receipts. I've been using it successfully for several years, and I want to update it to display personalization data for receipts when applicable. But I don't see anything in the API documentation about this. Nor do I see the data being returned for any receipts or transactions in my existing app.

Thank you in advance for your help.

**Comment by turboturtleD** [comment, marked as answer]
I think it shows up via “variations”.  Let me know if you can’t find it, and I’ll poke my app with a stick and offer further clues.

**Reply by Gareth064** [reply]
Yep, it just shows as one of the Variations.

Here it is in my DB

![image](https://user-images.githubusercontent.com/22307223/217645677-1e69f81b-7c80-4b80-bed8-3f2eb0c6035c.png)

And all I'm doing is looping the Variations on the Transaction to get this.

![image](https://user-images.githubusercontent.com/22307223/217645776-51b4ecbf-d063-4374-9dfd-2ebae0bedc50.png)

**Reply by joey-lott** [reply]
Thank you to you both for your helpful responses. That solved the problem for me. I appreciate that you took the time to respond.
